### ISSUE TITLE
[clsim] Implement correct wavelength acceptance for HQE DOMs (Trac #2162)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2154]
Currently HQE DOMs (mostly DeepCore) are simulated with 35% higher DOM efficiency compared to standard QE DOMs (IceCube). This is incorrect, as the wavelength acceptance is also different for DeepCore DOMs. (see attached plots). This wavelength dependence should be implemented in clsim. Work already done my A. Terliuk and C. Weaver towards this, but ticket was never officially created. Creating one now so that this does not remain on the back-burner - it is needed for next generation oscillation analyses. Mass (low energy) MC production on hold until this makes its way into tag of clsim.  

Migrated from https://code.icecube.wisc.edu/ticket/2162
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:15:23", 
    "description": "Currently HQE DOMs (mostly DeepCore) are simulated with 35% higher DOM efficiency compared to standard QE DOMs (IceCube). This is incorrect, as the wavelength acceptance is also different for DeepCore DOMs. (see attached plots). This wavelength dependence should be implemented in clsim. Work already done my A. Terliuk and C. Weaver towards this, but ticket was never officially created. Creating one now so that this does not remain on the back-burner - it is needed for next generation oscillation analyses. Mass (low energy) MC production on hold until this makes its way into tag of clsim.  ", 
    "reporter": "summer.blot", 
    "cc": "claudio.kopper, cweaver, jvansanten, dima", 
    "resolution": "fixed", 
    "_ts": "1550067323910946", 
    "component": "combo simulation", 
    "summary": "[clsim] Implement correct wavelength acceptance for HQE DOMs", 
    "priority": "critical", 
    "keywords": "HQE wavelength acceptance", 
    "time": "2018-06-12T11:26:39", 
    "milestone": "", 
    "owner": "amedina", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20180612T11:27:00: summer.blot changed attachment from "" to "DOM_acceptance.png"

**Comment by icecube-trac** [comment]
Trac update at 20180612T11:27:08: summer.blot changed attachment from "" to "HQE_wavelength acceptance.pdf"

**Comment by icecube-trac** [comment]
Trac update at 20180612T11:29:10: jvansanten commented:

> FYI: This is implemented in clsim/branches/multiprocess-server, which also addresses Geant4-induced GPU wastage by allowing multiple CPU processes to share a GPU.

**Comment by icecube-trac** [comment]
Trac update at 20180612T11:36:49: 
* andrii.terliuk changed owner from "andrii.terliuk" to "kjmeagher"
* andrii.terliuk changed status from "new" to "assigned"

**Comment by icecube-trac** [comment]
Trac update at 20180613T18:29:05: kjmeagher changed component from "icerec" to "simulation"

**Comment by icecube-trac** [comment]
Trac update at 20180922T08:47:23: olivas changed milestone from "" to "Winter Solstice 2018"

**Comment by icecube-trac** [comment]
Trac update at 20180922T08:47:56: olivas changed owner from "kjmeagher" to "olivas"

**Comment by icecube-trac** [comment]
Trac update at 20181002T15:16:44: 
* olivas changed status from "assigned" to "accepted"
* olivas changed summary from "Implement correct wavelength acceptance for HQE DOMs" to "[clsim] Implement correct wavelength acceptance for HQE DOMs"

**Comment by icecube-trac** [comment]
Trac update at 20181114T14:27:28: 
* olivas changed status from "accepted" to "assigned"
* olivas changed owner from "olivas" to "amedina"

**Comment by icecube-trac** [comment]
Trac update at 20181205T19:13:55: 
* amedina changed status from "assigned" to "closed"
* amedina commented:

> In changeset 167319:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="167319"
> fixes #2154
> ```


* amedina changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20181205T19:37:46: 
* jvansanten changed status from "closed" to "reopened"
* jvansanten changed resolution from "fixed" to ""
* jvansanten commented:

> I3CLSimPhotonToMCPEConverterForDOMs and setupDetector have no analogue in current trunk clsim, so simply merging them doesn't do anything. You will need some more refactoring to use them in classic clsim. Off the top of my head you would need to at least replace the setup steps of all the traysegments (I3CLSimMakeHits and friends) with setupDetector(), and make the I3PhotonToMCPEConverter module use I3CLSimPhotonToMCPEConverterForDOMs internally. While you're at it you might as well include the streaming MCPE compression step from I3CLSimClientModule, though the benefit might be reduced given that you have the entire I3PhotonSeriesMap in memory already.
> 
> If you have to do major surgery on the traysegments anyway, It might be simpler to just take all of branches/multiprocess-server.

**Comment by icecube-trac** [comment]
Trac update at 20181207T17:15:28: 
* olivas changed status from "reopened" to "closed"
* olivas changed resolution from "" to "duplicate"
* olivas commented:

> We're gonna back that latest change out and close this ticket as a duplicate of #2152, since this is solved on that branch already.  The multiprocess-server branch is the official future of clsim.

**Comment by icecube-trac** [comment]
Trac update at 20190112T02:41:16: 
* nwhitehorn commented:

> In changeset 170472:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="170472"
> fixes #2154
> ```


* nwhitehorn changed resolution from "duplicate" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:15:23: 
* olivas changed milestone from "Winter Solstice 2018" to ""
* olivas commented:

> Milestone Winter Solstice 2018 deleted
